tests/testclient.nim

```diff
@@ -1279,6 +1279,90 @@ suite "REST API client test suite":
     await server.closeWait()
 
+  asyncTest "Connection management test":
+    var router = RestRouter.init(testValidate)
+    router.api(MethodGet, "/test/conn") do (
+      ) -> RestApiResponse:
+      return RestApiResponse.response("ok")
+    router.api(MethodPost, "/test/conn") do (
+      contentBody: Option[ContentBody]) -> RestApiResponse:
+      return RestApiResponse.response("ok")
+
+    var sres = RestServerRef.new(router, serverAddress)
+    let server = sres.get()
+    server.start()
+
+    proc testGetConnection1(): RestPlainResponse {.
+      rest, endpoint: "/test/conn", meth: MethodGet.}
+    proc testGetConnection2(): RestPlainResponse {.
+      rest, endpoint: "/test/conn", meth: MethodGet,
+      connection: {}.}
+    proc testGetConnection3(): RestPlainResponse {.
+      rest, endpoint: "/test/conn", meth: MethodGet,
+      connection: {Dedicated}.}
+    proc testGetConnection4(): RestPlainResponse {.
+      rest, endpoint: "/test/conn", meth: MethodGet,
+      connection: {Close}.}
+    proc testGetConnection5(): RestPlainResponse {.
+      rest, endpoint: "/test/conn", meth: MethodGet,
+      connection: {Dedicated, Close}.}
+
+    proc testPostConnection1(body: string): RestPlainResponse {.
+      rest, endpoint: "/test/conn", meth: MethodPost.}
+    proc testPostConnection2(body: string): RestPlainResponse {.
+      rest, endpoint: "/test/conn", meth: MethodPost,
+      connection: {}.}
+    proc testPostConnection3(body: string): RestPlainResponse {.
+      rest, endpoint: "/test/conn", meth: MethodPost,
+      connection: {Dedicated}.}
+    proc testPostConnection4(body: string): RestPlainResponse {.
+      rest, endpoint: "/test/conn", meth: MethodPost,
+      connection: {Close}.}
+    proc testPostConnection5(body: string): RestPlainResponse {.
+      rest, endpoint: "/test/conn", meth: MethodPost,
+      connection: {Dedicated, Close}.}
+
+    template runGetTest(body: untyped, http: int, respbody: string,
+                        count: int): untyped =
+      var client = RestClientRef.new(serverAddress, HttpClientScheme.NonSecure)
+      try:
+        let response = await body(client)
+        check:
+          response.status == http
+          response.data.bytesToString() == respbody
+          client.session.connectionsCount == count
+      finally:
+        await client.closeWait()
+
+    template runPostTest(body: untyped, http: int, respbody: string,
+                         count: int): untyped =
+      var client = RestClientRef.new(serverAddress, HttpClientScheme.NonSecure)
+      try:
+        let resp = await body(client, "data")
+        check:
+          resp.status == http
+          resp.data.bytesToString() == respbody
+          client.session.connectionsCount == count
+      finally:
+        await client.closeWait()
+
+    try:
+      runGetTest(testGetConnection1, 200, "ok", 1)
+      runGetTest(testGetConnection2, 200, "ok", 1)
+      runGetTest(testGetConnection3, 200, "ok", 1)
```

Comment:
just for understanding: here, running the GET for connection3 twice would end up with 2 open connections, correct?

Reply:
Nope, pool of connections maintained by `HttpSession` which is part of `RestClientRef`, so if you will use single instance of `RestClientRef` and perform requests with either `{}` or `{Dedicated}` you will end up with 2 open connections, but if you close `RestClientRef` it will close all the connections, so does not matter how many times you perform `runGetTest(testGetConnection3)` it will always return `1`.